java/org/contikios/cooja/SimEventCentral.java

```diff
@@ -82,23 +82,14 @@ public String toString() {
     }
   }
   /** Help class for maintaining mote-specific observations */
-  private static class MoteObservation {
-    private final Mote mote;
-    private final Observable observable;
-    private final Observer observer;
-    public MoteObservation(Mote mote, Observable observable, Observer observer) {
+  private record MoteObservation(Mote mote, Observable observable, Observer observer) {
+    private MoteObservation(Mote mote, Observable observable, Observer observer) {
```

Comment:
I think this can be simplified as:
```java
private record MoteObservation(Mote mote, Observable observable, Observer observer) {
    public MoteObservation {
        observable.addObserver(observer);
    }
}
```

Reply:
That is even shorter, fixed!